graph-proxy/src/graphql/workflows.rs

```diff
@@ -56,22 +81,70 @@ enum WorkflowStatus {
     Errored(WorkflowErroredStatus),
 }
 
-impl TryFrom<IoArgoprojWorkflowV1alpha1WorkflowStatus> for WorkflowStatus {
-    type Error = WorkflowParsingError;
-
-    fn try_from(value: IoArgoprojWorkflowV1alpha1WorkflowStatus) -> Result<Self, Self::Error> {
+#[allow(clippy::missing_docs_in_private_items)]
+impl WorkflowStatus {
+    fn new(
+        value: IoArgoprojWorkflowV1alpha1WorkflowStatus,
+        metadata: Weak<Metadata>,
+    ) -> Result<Self, WorkflowParsingError> {
         match value.phase.as_deref() {
             Some("Pending") => Ok(Self::Pending(WorkflowPendingStatus::from(value))),
-            Some("Running") => Ok(Self::Running(WorkflowRunningStatus::try_from(value)?)),
-            Some("Succeeded") => Ok(Self::Succeeded(WorkflowSucceededStatus::try_from(value)?)),
-            Some("Failed") => Ok(Self::Failed(WorkflowFailedStatus::try_from(value)?)),
-            Some("Error") => Ok(Self::Errored(WorkflowErroredStatus::try_from(value)?)),
+            Some("Running") => Ok(Self::Running(WorkflowRunningStatus::new(value, metadata)?)),
+            Some("Succeeded") => Ok(Self::Succeeded(WorkflowSucceededStatus::new(
+                value, metadata,
+            )?)),
+            Some("Failed") => Ok(Self::Failed(WorkflowFailedStatus::new(value, metadata)?)),
+            Some("Error") => Ok(Self::Errored(WorkflowErroredStatus::new(value, metadata)?)),
             Some(_) => Err(WorkflowParsingError::UnrecognisedPhase),
             None => Err(WorkflowParsingError::UnrecognisedPhase),
         }
     }
 }
 
+lazy_static! {
+    static ref CLIENT: reqwest::Client = reqwest::Client::new();
+}
+
+#[allow(clippy::missing_docs_in_private_items)]
+macro_rules! fetch_tasks {
+    ($status:expr, $ctx:expr) => {{
```

Comment:
Since we're not generating the function declaration I think we can just use a function here. Alternatively we could create a macro to generate the entire `impl $iden`